core/api/src/main/java/com/blazebit/persistence/PagedArrayList.java

```diff
@@ -40,21 +38,38 @@
     private final int firstResult;
     private final int maxResults;
 
-    public PagedListImpl(KeysetPage keyset, long totalSize, int firstResult, int maxResults) {
+    /**
+     * Constructs a new empty paged array list.
+     *
+     * @param keyset      The keyset page for this page
+     * @param totalSize   The total size of the result
+     * @param firstResult The first result index within the overall result
+     * @param maxResults  The maximum result count for a page
+     */
+    public PagedArrayList(KeysetPage keyset, long totalSize, int firstResult, int maxResults) {
         this.keyset = keyset;
         this.totalSize = totalSize;
         this.page = (int) Math.floor((firstResult == -1 ? 0 : firstResult) * 1d / maxResults) + 1;
-        this.totalPages = (int) Math.ceil(totalSize * 1d / maxResults);
+        this.totalPages = totalSize < 1 ? (int) totalSize : (int) Math.ceil(totalSize * 1d / maxResults);
         this.firstResult = firstResult;
         this.maxResults = maxResults;
     }
 
-    PagedListImpl(Collection<? extends T> collection, KeysetPage keyset, long totalSize, int firstResult, int maxResults) {
+    /**
+     * Constructs a new paged array list from the given collection.
+     *
+     * @param collection  The collection of elements for this page
+     * @param keyset      The keyset page for this page
+     * @param totalSize   The total size of the result
+     * @param firstResult The first result index within the overall result
+     * @param maxResults  The maximum result count for a page
+     */
+    public PagedArrayList(Collection<? extends T> collection, KeysetPage keyset, long totalSize, int firstResult, int maxResults) {
```

Comment:
Why don't you share the common constructor code?

Reply:
Because I want to make use of the super constructor call.